lib/ecto/uuid.ex

```diff
@@ -29,9 +29,9 @@ defmodule Ecto.UUID do
   catch
     :error -> :error
   else
-    casted ->
-      {:ok, casted}
+    casted -> {:ok, casted}
   end
+  def cast(<< _::128 >> = binary), do: binary |> encode() |> cast()
```

Comment:
@michalmuskala I agree, but doing that will bypass the validations from `cast_version` and `cast_variant`. Do you think that's ok?